### PR TITLE
KevinC/ add logout modal

### PR DESCRIPTION
**Pull request by kevinchen220**
## Notion ticket link

[Implement Log Out Modal and Flow](https://www.notion.so/uwblueprintexecs/Implement-Log-Out-Modal-and-Flow-0119c7c9c45b4d57a82db4493d04b6fb?pvs=4)

## Implementation description
* Added a Log out modal using `ModalComponent`
* Log out modal is controlled by the `Logout` button
* Moved to previous logout function to the confirmation button onClick in the modal
  * Not exactly sure how to test that

## Steps to test
1. Click logout button on the top right of the dashboard
2. See the Logout Modal that pops up

## What should reviewers focus on?
* Whether the logout function is correct

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have run the appropriate linter(s)
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

image

### CODE REVIEW

File: frontend/src/components/common/ModalComponent.tsx [hunk 1]
```diff
@@ -46,6 +47,7 @@ const ModalComponent = ({
   onLeftButtonClick,
   titleColor,
   mainButtonVariant,
+  showModalCloseButton = true,
```
Comment (jess94yng) [inline frontend/src/components/common/ModalComponent.tsx:50]: Currently when the log out modal first pops open, the cancel button is in a selected state (blue ring around it) - see if that can be fixed?

**Review by jess94yng — CHANGES_REQUESTED** [review]

File: frontend/src/components/auth/Logout.tsx [hunk 2]
```diff
@@ -1,33 +1,47 @@
...
+import React, { useContext, useState } from "react";
 import { LogOut } from "react-feather";
 
 import authAPIClient from "../../APIClients/AuthAPIClient";
 import AuthContext from "../../contexts/AuthContext";
+import LogoutModal from "./LogoutModal";
 
 const LogoutButton = (): React.ReactElement => {
   const { authenticatedUser, setAuthenticatedUser } = useContext(AuthContext);
+  const [openModal, setOpenModal] = useState(false);
```
Comment (bowenzhu1) [inline frontend/src/components/auth/Logout.tsx:11]: we could use `useDisclosure` instead here which would be a bit cleaner. - to make use of ChakraUI modal handling logic. you can cmd+shift+f search for other files where we use it for reference. but other than that lgtm

File: frontend/src/components/common/ModalComponent.tsx (hunk 1 above)
Comment (kevinchen220) [reply frontend/src/components/common/ModalComponent.tsx:50]: Looks like that's the default behaviour of ChakraUI modals, the first tab-able element is selected. Not sure if there is an easy fix

Comment (kevinchen220) [reply frontend/src/components/common/ModalComponent.tsx:50]: Nevermind I got it to work

**Review by bowenzhu1 — APPROVED** [review]
lgtm to merge

**Review by jess94yng — APPROVED** [review]

**Pull request merged**